frontend/src/journal.ts

```diff
@@ -79,14 +83,27 @@ export class FavaJournal extends SortableJournal {
       // Filter for payees when clicking on them.
       // Note: any special characters in the payee string are escaped so the
       // filter matches against the payee literally.
-      addFilter(`payee:"${target.innerText}"`, true);
+      addFilter(`payee:"^${escape(target.innerText)}$"`);
+    } else if (target.tagName === "DT") {
+      // Filter for metadata key when clicking on the key. The key tag text
+      // includes the colon.
+      const expr = `${target.innerText}""`;
+      if (target.className === "metadata-entry") {
```

Comment:
I think we can do without the extra classes by using the following check here:

    if (!target.closest('.postings')) {

Reply:
That's clever! I undid the change to the template and updated these checks.